packages/js/src/types.ts

```diff
@@ -99,6 +118,29 @@ export type Cta = {
   action?: MessageAction;
 };
 
+export type WorkflowInfo = {
+  _id: string;
+  name: string;
+  critical: boolean;
+  tags?: string[];
+  identifier: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data?: Record<string, any>;
+};
+
+export type ChannelPreference = {
+  email?: boolean;
+  sms?: boolean;
+  in_app?: boolean;
+  chat?: boolean;
+  push?: boolean;
+};
+
+export type ChannelPreferenceOverride = {
```

Comment:
This seems overly complicated for the current frontend needs and is driven by the current state of the API. Let's build the DX on an ideal scenario and hide the API tech-debt with internal helper methods. Then we will update the API.

Reply:
I think that we can remove it from the new Preference class for now, not even sure if that is used often by clients.